src/Files.App.Storage/NativeStorage/NativeFolder.cs

```diff
@@ -0,0 +1,204 @@
+﻿using Files.Sdk.Storage;
+using Files.Sdk.Storage.Enums;
+using Files.Sdk.Storage.Extensions;
+using Files.Sdk.Storage.LocatableStorage;
+using Files.Sdk.Storage.ModifiableStorage;
+using Files.Sdk.Storage.MutableStorage;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Files.App.Storage.NativeStorage
+{
+    /// <inheritdoc cref="IFolder"/>
+    public sealed class NativeFolder : NativeStorable, ILocatableFolder, IModifiableFolder, IMutableFolder
+    {
+        public NativeFolder(string path)
+            : base(path)
+        {
+        }
+
+        /// <inheritdoc/>
+        public Task<IFile> GetFileAsync(string fileName, CancellationToken cancellationToken = default)
+        {
+            var path = System.IO.Path.Combine(Path, fileName);
+
+            if (!File.Exists(path))
+                throw new FileNotFoundException();
+
+            return Task.FromResult<IFile>(new NativeFile(path));
+        }
+
+        /// <inheritdoc/>
+        public Task<IFolder> GetFolderAsync(string folderName, CancellationToken cancellationToken = default)
+        {
+            var path = System.IO.Path.Combine(Path, folderName);
+
+            if (!Directory.Exists(path))
+                throw new FileNotFoundException();
+
+            return Task.FromResult<IFolder>(new NativeFolder(path));
+        }
+
+        /// <inheritdoc/>
+        public async IAsyncEnumerable<IStorable> GetItemsAsync(StorableKind kind = StorableKind.All, [EnumeratorCancellation] CancellationToken cancellationToken = default)
+        {
+            if (kind == StorableKind.Files)
+            {
+                foreach (var item in Directory.EnumerateFiles(Path))
+                    yield return new NativeFile(item);
+            }
+            else if (kind == StorableKind.Folders)
+            {
+                foreach (var item in Directory.EnumerateDirectories(Path))
+                    yield return new NativeFolder(item);
+            }
+            else
+            {
+                foreach (var item in Directory.EnumerateFileSystemEntries(Path))
+                {
+                    if (File.Exists(item))
+                        yield return new NativeFile(item);
+                    else
+                        yield return new NativeFolder(item);
+                }
+            }
+
+            await Task.CompletedTask;
+        }
+
+        /// <inheritdoc/>
+        public Task DeleteAsync(IStorable item, bool permanently = false, CancellationToken cancellationToken = default)
+        {
+            _ = permanently;
```

Comment:
Looking at this section, the `permanently` has no use in the method and could be removed.

Reply:
This method is part of IModifiableFolder interface